fish-rust/src/builtins/functions.rs

```diff
@@ -0,0 +1,413 @@
+                streams.out.append(sprintf!("%ls\n", "stdin"));
+            }
+            if opts.verbose {
+                if props.copy_definition_file.is_some() {
+                    if let Some(df) = &props.definition_file {
+                        streams.out.append(sprintf!("%ls\n", df));
+                    } else {
+                        streams.out.append(sprintf!("%ls\n", "stdin"));
+                    }
+                } else {
+                    streams.out.append(sprintf!(
+                        "%ls\n",
+                        if props.is_autoload.load() {
+                            "autoloaded"
+                        } else {
+                            "n/a"
+                        }
+                    ));
+                }
+                streams.out.append(sprintf!(
+                    "%d\n%ls\n",
+                    props.definition_lineno(),
+                    if props.shadow_scope {
+                        "scope-shadowing"
+                    } else {
+                        "no-scope-shadowing"
+                    }
+                ));
+                if !props.description.is_empty() {
+                    streams.out.append(sprintf!(
+                        "%ls\n",
+                        escape_string(
+                            &props.description,
+                            EscapeStringStyle::Script(
+                                EscapeFlags::NO_PRINTABLES | EscapeFlags::NO_QUOTED
+                            )
+                        ),
+                    ));
+                }
+            }
+        } else if opts.verbose {
+            streams.out.append(L!("n/a\nn/a\n0n/a\nn/a\n"));
+        } else {
+            streams.out.append(L!("n/a\n"));
+        }
+        // Historical - this never failed?
+        return STATUS_CMD_OK;
+    }
+
+    if opts.handlers {
+        if !opts.handlers_type.is_empty()
+            && !event::EVENT_FILTER_NAMES.contains(&opts.handlers_type.as_utfstr())
+        {
+            streams.err.append(wgettext_fmt!(
+                "%ls: Expected generic | variable | signal | exit | job-id for --handlers-type\n",
+                cmd
+            ));
+            return STATUS_INVALID_ARGS;
+        }
+        event::print(streams, &opts.handlers_type);
+        return STATUS_CMD_OK;
+    }
+
+    if opts.query && args.is_empty() {
+        return STATUS_CMD_ERROR;
+    }
+
+    if opts.list || args.is_empty() {
+        let mut names = function::get_names(opts.show_hidden);
+        names.sort();
+        if !streams.out_is_redirected && unsafe { isatty(STDOUT_FILENO) == 1 } {
+            let mut buff = WString::new();
+            let mut first: bool = true;
+            for name in names {
+                if !first {
+                    buff.push_utfstr(L!(", "));
+                }
+                buff.push_utfstr(&name);
+                first = false;
+            }
+            // TODO: reformat_for_screen
```

Comment:
See the TODO - this would have to be implemented before it can be merged.

(not that the reformatting or even the ","-separated format is all that great - I consider it a mistake, tbh)

Reply:
Turns out it's easier than I thought because reformat_for_screen was already ported.